frontend/packages/core/src/Charts/timeseries.tsx

```diff
@@ -0,0 +1,163 @@
+  // The assumption is that multiple lines would want to share the same Y Axis.
+  drawDots?: boolean;
+  enableLegend?: boolean;
+  enableGrid?: boolean;
+  tickFormatterFunc?: (timeStamp: number) => string;
+  xDomainSpread?: number | null;
+  yDomainSpread?: number | null;
+  connectNulls?: boolean;
+  regularIntervalTicks?: boolean;
+  // TODO: add ref dots, ref areas, zoom enabled, auto colors,
+  // tooltip options, activeDot options, dark mode / styling,
+}
+
+/*
+  For the lines, use the `dataKey` property to denote which data points belong to that line. Make sure that
+  all the dataKeys match appropriately (the XAxis datakey should correspond to the data that is graphed along
+  the XAxis, and same for Y. Note that we currently set the XAxis to be a time scale, hence the name
+  Timeseries Chart).
+  
+  The data will be internally sorted by the XAxis dataKey when using single line mode. When having multiple lines
+  (singleLineMode is false) then the user is responsible for sorting and grouping the data appropriately. 
+  If they do not sort and group it, there can be discontinuities in the lines. Also, when using multiple lines,
+  the user should pass the yAxisDataKey as the biggest ranged y axis datakey in the data, otherwise data will get chopped off.
+
+  The timestamps are interpreted as unix milliseconds.
+*/
+const TimeseriesChart = ({
+  data,
+  xAxisDataKey = "timestamp",
+  yAxisDataKey = "value",
+  lines,
+  refLines,
+  singleLineMode = true,
+  drawDots = true,
+  enableLegend = true,
+  enableGrid = true,
+  tickFormatterFunc = localTimeFormatter,
+  xDomainSpread = 0.2,
+  yDomainSpread = 0.2,
+  connectNulls = false,
+  regularIntervalTicks = false,
+}: TimeseriesChartProps) => {
+  if (singleLineMode) {
+    data.sort((a, b) => a[xAxisDataKey] - b[xAxisDataKey]);
+  }
+  const [yAxisDomainMin, yAxisDomainMax] = calculateDomainEdges(
+    data,
+    yAxisDataKey,
+    yDomainSpread ?? 0
+  );
+  const [xAxisDomainMin, xAxisDomainMax] = calculateDomainEdges(
+    data,
+    xAxisDataKey,
+    xDomainSpread ?? 0
+  );
+
+  // In the spirit of making a friendly UX, there is an option to generate evenly spaced, round-timestamped, ticks.
+  // Depending on the distance between the max and min timestamps, we calculate a set of ticks at certain intervals
+  // (I.e. if there are several minutes, we might use minute intervals, whereas if there are only a few minutes range,
+  // we would use 30 second intervals, and if the range consists of hours, we might use 15 or 30 minute intervals)
+  let ticks = [];
+  if (regularIntervalTicks) {
+    ticks = calculateTicks(data, xAxisDataKey);
+  }
+
+  return (
+    <ResponsiveContainer width="100%" height="100%">
+      <LineChart data={data}>
+        {enableGrid ? <CartesianGrid /> : null}
+        <XAxis
+          dataKey={xAxisDataKey}
+          type="number"
+          domain={[xAxisDomainMin, xAxisDomainMax]}
+          tickFormatter={tickFormatterFunc || null}
```

Comment:
not sure how this could be null since it's a required prop. is this superfluous? 
```suggestion
          tickFormatter={tickFormatterFunc}
```